File: src/Web/Client/Services/WorkflowsService.cs
using Flurl;
using MudBlazor;
using Shared.Dtos;
using Shared.ViewModels;

namespace Web.Client.Services;

public class WorkflowsService : ApiService
{
    public WorkflowsService(HttpClient httpClient, ISnackbar snackbar) 
        : base(httpClient, snackbar)
    {
    }

    public async Task<WorkflowVM?> Get(Guid projectId)
    {
        var url = "workflows"
            .SetQueryParam("projectId", projectId)
            .ToString();
        return await Get<WorkflowVM>(url);
    }

    public async Task<bool> AddStatus(Guid workflowId, Guid projectId, AddWorkflowStatusDto model)
        => await Post($"workflows/{workflowId}/statuses", model, ProjectIdHeader(projectId));

    public async Task<bool> AddTransition(Guid workflowId, Guid projectId, AddWorkflowTransitionDto model)
        => await Post($"workflows/{workflowId}/transitions", model, ProjectIdHeader(projectId));

    public async Task<bool> DeleteStatus(Guid workflowId, Guid statusId, Guid projectId)
        => await Post($"workflows/{workflowId}/statuses/{statusId}/delete", ProjectIdHeader(projectId));

    public async Task<bool> DeleteTransition(Guid workflowId, Guid projectId, DeleteWorkflowTransitionDto model)
        => await Post($"workflows/{workflowId}/transitions/delete", model, ProjectIdHeader(projectId));

    public async Task<bool> ChangeInitialStatus(Guid workflowId, Guid projectId, ChangeInitialWorkflowStatusDto model)
        => await Post($"workflows/{workflowId}/change-initial-status", model, ProjectIdHeader(projectId));

    private static Headers ProjectIdHeader(Guid projectId)
        => Headers.From(("ProjectId", projectId.ToString()));
}

File: src/Web/Client/composables/workflowsService.ts
import type { AddWorkflowStatusDto, AddWorkflowTransitionDto, ChangeInitialWorkflowStatusDto, DeleteWorkflowStatusDto, DeleteWorkflowTransitionDto } from "~/types/dtos/workflows";
import type { WorkflowVM } from "~/types/viewModels/workflows";

export const useWorkflowsService = () => {
    const api = useApi();

    return {
        async getWorkflow(projectId: string) {
            return await api.sendGetRequest<WorkflowVM>(`workflows?projectId=${projectId}`);
        },
        async addStatus(id: string, projectId: string, model: AddWorkflowStatusDto) {
            await api.sendPostRequest(`workflows/${id}/statuses`, model, { 'ProjectId': projectId });
        },
        async addTransition(id: string, projectId: string, model: AddWorkflowTransitionDto) {
            await api.sendPostRequest(`workflows/${id}/transitions`, model, { 'ProjectId': projectId });
        },
        async deleteStatus(id: string, projectId: string, model: DeleteWorkflowStatusDto) {
            await api.sendPostRequest(`workflows/${id}/statuses/delete`, model, { 'ProjectId': projectId });
        },
        async deleteTransition(id: string, projectId: string, model: DeleteWorkflowTransitionDto) {
            await api.sendPostRequest(`workflows/${id}/transitions/delete`, model, { 'ProjectId': projectId });
        },
        async changeInitialStatus(id: string, projectId: string, model: ChangeInitialWorkflowStatusDto) {
            await api.sendPostRequest(`workflows/${id}/initial-status`, model, { 'ProjectId': projectId });
        }
    }
}